### PR TITLE
Modify Signals to render a HTML button without React.

### PR DESCRIPTION
**Pull request by adpatter**
This PR is in reference to https://github.com/jupyterlab/extension-examples/issues/144.

Modify the Signals example to create an HTML button instead of a React button.  I think this modification will make the documentation easier to follow for persons who don't know React.

Further, the Lumino Widgets were modified in order to produce the desired result shown in preview.png.  The result of the current React implementation doesn't actually produce what is in the preview image.

CSS class naming conventions were followed.

### CODE REVIEW

**Comment by meeseeksmachine** [comment]
This pull request has been mentioned on **Jupyter Community Forum**. There might be relevant details there:

https://discourse.jupyter.org/t/is-usage-of-react-necessary-for-writing-jupyterlab-extensions-effectively/7825/6

**Comment by adpatter** [comment]
Can anyone help me understand why the PR is failing 2 Checks?

**Comment by meeseeksmachine** [comment]
This pull request has been mentioned on **Jupyter Community Forum**. There might be relevant details there:

https://discourse.jupyter.org/t/can-anyone-help-me-understand-why-my-pr-is-failing-2-checks/7932/1

**Comment by fcollonval** [comment]
Thanks @adpatter 

The errors come from the snippets in the README. To ensure the code and the README are aligned we are using a tool call [embedme](https://github.com/zakhenry/embedme) that copy the code snippets within the README. And in the CI, we are checking the code and snippets are aligned.

To update the README, you need to execute `jlpm run embedme` in the examples root folder. Then you can push the updated README.

More details - in the CI logs you can see:

```
  Analysing basics/signals/README.md...
   basics/signals/README.md#L26-L30 Embedded 1 lines from file src/button.ts#L36-L36
   basics/signals/README.md#L34-L40 Embedded 3 lines from file src/button.ts#L15-L17
   basics/signals/README.md#L45-L49 Embedded 1 lines from file src/panel.ts#L29-L29
   basics/signals/README.md#L53-L57 Embedded 1 lines from file src/button.ts#L28-L28
   basics/signals/README.md#L72-L92 Embedded 17 lines from file src/button.ts#L8-L24
   basics/signals/README.md#L98-L104 Embedded 3 lines from file src/button.ts#L26-L28
   basics/signals/README.md#L109-L113 Embedded 1 lines from file src/button.ts#L30-L30
   basics/signals/README.md#L124-L131 No changes required, already up to date
   basics/signals/README.md#L139-L155 Embedded 13 lines from file src/panel.ts#L18-L30
   basics/signals/README.md#L159-L163 Embedded 1 lines from file src/panel.ts#L29-L29
   basics/signals/README.md#L178-L182 Embedded 1 lines from file src/panel.ts#L32-L32
   basics/signals/README.md#L187-L199 Embedded 9 lines from file src/panel.ts#L32-L40
```

So all snippets in the README needs to be updated except the one in lines 124-131.

File: basics/signals/README.md [hunk 1]
```diff
@@ -52,86 +51,83 @@ this._widget.stateChanged.connect(this._logMessage, this);
...
+    this.node.addEventListener('click', () => {
+      this._count.clickCount = this._count.clickCount + 1;
+      this._stateChanged.emit(this._count);
+    });
+  }
+  //...
 }
 ```
 
 `ButtonWidget` also contain a private attribute `_count` of type `ICount`.
```
Comment (fcollonval) [inline basics/signals/README.md:96]:
```suggestion
`ButtonWidget` also contains a private attribute `_count` of type `ICount`.
```

Taking the chance to correct that one 😉

File: basics/signals/package.json [hunk 2]
```diff
@@ -45,13 +45,16 @@
...
     "@jupyterlab/launcher": "^3.0.0-rc.15",
     "@jupyterlab/mainmenu": "^3.0.0-rc.15",
     "@jupyterlab/translation": "^3.0.0-rc.15",
     "@lumino/algorithm": "^1.3.3",
     "@lumino/coreutils": "^1.5.3",
     "@lumino/datagrid": "^0.3.1",
-    "@lumino/disposable": "^1.4.3"
+    "@lumino/disposable": "^1.4.3",
+    "@lumino/signaling": "^1.3.3",
+    "@lumino/widgets": "^1.3.3"
```
Comment (fcollonval) [inline basics/signals/package.json:57]:
```suggestion
    "@lumino/widgets": "^1.16.1"
```

Alignment with jupyterlab dependency constrain

File: basics/signals/package.json [hunk 3]
```diff
@@ -45,13 +45,16 @@
   },
   "dependencies": {
     "@jupyterlab/application": "^3.0.0-rc.15",
+    "@jupyterlab/apputils": "^3.0.3",
     "@jupyterlab/launcher": "^3.0.0-rc.15",
     "@jupyterlab/mainmenu": "^3.0.0-rc.15",
     "@jupyterlab/translation": "^3.0.0-rc.15",
```
Comment (fcollonval) [inline basics/signals/package.json:51]:
```suggestion
    "@jupyterlab/application": "^3.0.3",
    "@jupyterlab/apputils": "^3.0.3",
    "@jupyterlab/launcher": "^3.0.3",
    "@jupyterlab/mainmenu": "^3.0.3",
    "@jupyterlab/translation": "^3.0.3",
```

it is a good practice to be consistent on the jupyterlab dependencies.

File: basics/signals/src/panel.ts [hunk 4]
```diff
@@ -3,24 +3,23 @@ import {
...
 } from '@jupyterlab/translation';
-import { StackedPanel } from '@lumino/widgets';
+
+import { Panel } from '@lumino/widgets';
 
 import { ButtonWidget, ICount } from './button';
-/**
- * The class name added to console panels.
- */
-const PANEL_CLASS = 'jp-tutorial-view';
```
Comment (fcollonval) [inline basics/signals/src/panel.ts:12]: Could you keep the class name as constant as this is to align with JupyterLab practice?

File: basics/signals/src/button.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  constructor(options = { node: document.createElement('button') }) {
+    super(options);
+
+    this.node.textContent = 'Clickme';
+
+    /**
+     * The class name, jp-ButtonWidget, follows the CSS class naming
+     * convention for classes that extend lumino.Widget.
+     */
+    this.addClass('jp-ButtonWidget');
```
Comment (fcollonval) [inline basics/signals/src/button.ts:18]: Could you extract the class name as constant to align with JupyterLab practice?

File: basics/signals/src/button.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+export interface ICount {
+  clickCount: number;
+}
+
+export class ButtonWidget extends Widget {
+  constructor(options = { node: document.createElement('button') }) {
+    super(options);
+
+    this.node.textContent = 'Clickme';
```
Comment (fcollonval) [inline basics/signals/src/button.ts:12]: What about

```suggestion
    this.node.textContent = 'Click me';
```

**Review by fcollonval — CHANGES_REQUESTED** [review]
@adpatter I added some comments and suggestions.

**Comment by adpatter** [comment]
@fcollonval Thank you for reviewing my code and providing guidance.  In particular, thank you for the guidance on how to choose dependency versions - it makes sense to me now.  I made all of the requested corrections to the code.

It looks like the PR has passed all the checks.  Thank you.

File: basics/signals/README.md (hunk 1 above)
Comment (adpatter) [reply basics/signals/README.md]: Fixed as requested.

File: basics/signals/package.json (hunk 2 above)
Comment (adpatter) [reply basics/signals/package.json]: Fixed as requested.  Thanks for the guidance.

File: basics/signals/package.json (hunk 3 above)
Comment (adpatter) [reply basics/signals/package.json]: Fixed as requested.

File: basics/signals/src/panel.ts (hunk 4 above)
Comment (adpatter) [reply basics/signals/src/panel.ts:12]: Fixed as requested.  However, I noticed that the documentation seemed to indicated that they were moving away from that practice: "CSS classnames are defined inline in the code. We used to put them as all caps file-level consts, but we are moving away from that." (https://jupyterlab.readthedocs.io/en/stable/developer/css.html)

File: basics/signals/src/button.ts (hunk 5 above)
Comment (adpatter) [reply basics/signals/src/button.ts]: Fixed as requested.

File: basics/signals/src/button.ts (hunk 6 above)
Comment (adpatter) [reply basics/signals/src/button.ts]: Fixed as requested.

**Comment by adpatter** [comment]
A .vscode file snuck in there, so I resubmitted the PR.

**Comment by adpatter** [comment]
@fcollonval It took me awhile to figure out how embedme works.  However, I think I have all the mappings correct now - I apologize if this caused you any inconvenience.  I will be more careful about that in the future.  Thank you.

File: basics/signals/src/panel.ts (hunk 4 above)
Comment (fcollonval) [reply basics/signals/src/panel.ts:12]: Thanks for pointing this out - I miss that information. I have opened a new issue to conform to that principle. So we can merge this one.

**Review by fcollonval — APPROVED** [review]
LGTM ! Thanks @adpatter

**Pull request merged**